azure_data_cosmos_engine/src/query/producer/mod.rs

```diff
@@ -100,15 +151,23 @@ impl ItemProducer {
         Self::NonStreaming(NonStreamingStrategy::new(pkranges, sorting))
     }
 
+    /// Creates a producer for read many operations.
+    ///
+    /// This strategy processes query chunks sequentially, where each chunk will map to its own partition key range.
+    /// It will exhaust one chunk completely before moving to the next.
+    pub fn read_many(query_chunks: Vec<HashMap<String, Vec<(usize, String, String)>>>) -> Self {
+        Self::ReadMany(ReadManyStrategy::new(query_chunks))
+    }
+
     /// Gets the [`DataRequest`]s that must be performed in order to add additional data to the partition buffers.
-    pub fn data_requests(&mut self) -> Vec<DataRequest> {
+    pub fn data_requests(&mut self) -> crate::Result<Vec<DataRequest>> {
```

Comment:
Why make this change?

Reply:
was cherry-picking changes from Ashley's hybrid search PR - it is to accommodate for the Hybrid state being introduced since it doesn't follow the same logic of using `Ok()` on the other strategies' requests() method, and as a result the outer method has to return the same type - @analogrelay do correct me if I'm wrong